bot/exts/moderation/infraction/_utils.py

```diff
@@ -188,6 +190,40 @@ async def notify_infraction(
     return await send_private_embed(user, embed)
 
 
+async def notify_infraction(
+        infraction: Infraction,
+        user: t.Optional[UserSnowflake] = None,
+        reason: t.Optional[str] = None
+) -> bool:
+    """
+    DM a user about their new infraction and return True if the DM is successful.
+
+    `user` and `reason` can be used to override what is in `infraction`. Otherwise, this data will
+    be retrieved from `infraction`.
+
+    Also return False if the user needs to be fetched but fails to be fetched.
+    """
+    if user is None:
+        user = discord.Object(infraction["user"])
+
+    # Sometimes user is a discord.Object; make it a proper user.
+    try:
+        if not isinstance(user, (discord.Member, discord.User)):
+            user = await bot.instance.fetch_user(user.id)
+    except discord.HTTPException as e:
+        log.error(f"Failed to DM {user.id}: could not fetch user (status {e.status})")
+        return False
+
+    type_ = infraction["type"].replace("_", " ").title()
+    icon = INFRACTION_ICONS[infraction["type"]][0]
+    expiry = time.format_infraction_with_duration(infraction["expires_at"])
+
+    if reason is None:
+        reason = infraction["reason"]
```

Comment:
Pretty sure that will never happen because of `reason` coming from discord.py and there an empty reason would become the default (None).

When is an empty string and None not functionally equivalent?

Reply:
It is possible by invoking it as follows: `!command ""`